torch/fx/graph.py

```diff
@@ -337,15 +337,19 @@ def type_repr(o : Any):
 
             typename = _type_repr(o)
 
-            # This is a generic type, e.g. typing.List[torch.Tensor]
             if hasattr(o, '__origin__'):
+                # This is a generic type, e.g. typing.List[torch.Tensor]
                 origin_type = _origin_type_map.get(o.__origin__, o.__origin__)
                 origin_typename = add_global(_type_repr(origin_type), origin_type)
 
-                # Assign global names for each of the inner type variables.
-                args = [type_repr(arg) for arg in o.__args__]
+                if hasattr(o, '__args__'):
+                    # Assign global names for each of the inner type variables.
+                    args = [type_repr(arg) for arg in o.__args__]
 
-                return f'{origin_typename}[{",".join(args)}]'
+                    return f'{origin_typename}[{",".join(args)}]'
+                else:
```

Comment:
unreachable?

Reply:
No, different Python versions have different behavior. In my tests on Python 3.9, a bare `typing.Tuple` type expression does not have an attribute `__args__`. Pre-3.9 has attribute `__args__`, but it is empty. We will need to keep both this branch and your suggested branch above

Reply:
3.9:

```
Python 3.9.5 (default, Jun  4 2021, 12:28:51) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Tuple
>>> t = Tuple
>>> hasattr(t, '__args__')
False
```

3.8:

```
Python 3.8.12 (default, Oct 12 2021, 13:49:34) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from typing import Tuple
>>> t = Tuple
>>> hasattr(t, '__args__')
True
>>> t.__args__
()
```